### PR TITLE
Add dbt cloud connector

### PR DESCRIPTION
**Pull request by mars-lan**
### 🤔 Why?

To extract dbt-related metadata directly from [dbt Cloud](https://docs.getdbt.com/docs/dbt-cloud/cloud-overview).

### 🤓 What?

This connector is essentially a wrapper around the [dbt core connector](https://github.com/MetaphorData/connectors/blob/main/metaphor/dbt/README.md). It uses [dbt Cloud Admin API](https://docs.getdbt.com/docs/dbt-cloud/dbt-cloud-api/admin-cloud-api) to download `manifest.json` generated from the latest successful run for a particular job and pass the file to dbt core connector for further processing.

### 🧪 Tested?

Added unit tests & manually verified.

### CODE REVIEW

File: metaphor/dbt/cloud/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+```yaml
+account_id: <dbt_cloud_account_id>
+job_id: <dbt_cloud_job_id>
+service_token: <service_account_token>
+output:
+  file:
+    directory: <output_directory>
+```
+
+You can extract `account_id` & `job_id` from a particular dbt job's URL, which has the format `https://cloud.getdbt.com/#/accounts/<account_id>/projects/<project_id>/jobs/<job_id>/`.
```
Comment (elic-eon) [inline metaphor/dbt/cloud/README.md:24]: Will an account have many jobs and do we need to have many crawlers run parallel?

File: metaphor/dbt/cloud/extractor.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+    def get_last_successful_run(self, job_id: int) -> Tuple[int, int]:
+        """Get the project and run IDs of the last successful run for a job"""
+
+        offset = 0
+        page_size = 50
+        while True:
+            # https://docs.getdbt.com/dbt-cloud/api-v2#operation/listRunsForAccount
+            resp = self._get(
+                "runs/", {"order_by": "-id", "limit": page_size, "offset": offset}
```
Comment (elic-eon) [inline metaphor/dbt/cloud/extractor.py:52]: Looks [listRunsForAccount](https://docs.getdbt.com/dbt-cloud/api-v2#operation/listRunsForAccount) provides an optional job_definition_id for filtering jobId. Should we use it?

Comment (mars-lan) [reply metaphor/dbt/cloud/extractor.py]: Good point. Use `job_definition_id` to filter instead.

File: metaphor/dbt/cloud/README.md (hunk 1 above)
Comment (mars-lan) [reply metaphor/dbt/cloud/README.md:24]: A job maps to a particular dbt project in the repo so typically there's only one. If there are multiple dbt projects, it's probably better to set up separate connectors than to mix the output into one.

**Review by elic-eon — APPROVED** [review]

**Pull request merged**